internal/parser/simple_parser_rules.go

```diff
@@ -51,23 +51,37 @@ func (p *simpleParser) parseSQLStatement(r reporter) (stmt *ast.SQLStmt) {
 	case token.KeywordCommit:
 		stmt.CommitStmt = p.parseCommitStmt(r)
 	case token.KeywordCreate:
-		p.parseCreateStmt(stmt, r)
+		p.parseCreateStmts(stmt, r)
 	case token.KeywordDelete:
-		stmt.DeleteStmt = p.parseDeleteStmt(r)
+		p.parseDeleteStmts(stmt, nil, r)
 	case token.KeywordDetach:
 		stmt.DetachStmt = p.parseDetachDatabaseStmt(r)
+	case token.KeywordDrop:
+		p.parseDropStmts(stmt, r)
 	case token.KeywordEnd:
 		stmt.CommitStmt = p.parseCommitStmt(r)
+	case token.KeywordInsert:
+		stmt.InsertStmt = p.parseInsertStmt(nil, r)
+	case token.KeywordReindex:
+		stmt.ReIndexStmt = p.parseReIndexStmt(r)
+	case token.KeywordRelease:
+		stmt.ReleaseStmt = p.parseReleaseStmt(r)
+	case token.KeywordReplace:
+		stmt.InsertStmt = p.parseInsertStmt(nil, r)
 	case token.KeywordRollback:
 		stmt.RollbackStmt = p.parseRollbackStmt(r)
+	case token.KeywordSavepoint:
+		stmt.SavepointStmt = p.parseSavepointStmt(r)
 	case token.KeywordSelect:
-		stmt.SelectStmt = p.parseSelectStmt(r)
+		stmt.SelectStmt = p.parseSelectStmt(nil, r)
+	case token.KeywordUpdate:
+		p.parseUpdateStmts(stmt, nil, r)
 	case token.KeywordVacuum:
 		stmt.VacuumStmt = p.parseVacuumStmt(r)
 	case token.KeywordValues:
-		stmt.SelectStmt = p.parseSelectStmt(r)
+		stmt.SelectStmt = p.parseSelectStmt(nil, r)
```

Comment:
What does `nil` as an argument tell us? What does it mean? Is that necessary? Or can we add another `parseSelect` method with a different set of parameters?

Reply:
`nil` parameter allows us to manage all types of select stmts that we see. With or without the `With Clause`, starting with different keywords etc